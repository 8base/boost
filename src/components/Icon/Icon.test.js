// @flow
import React from 'react';
import { Icon } from './';

describe('<Icon />', () => {
  it('should shallow input', () => {
    const wrapper = mount(
      <Icon name="Alert" color="DANGER" title="Alert icon" />,
    );

    expect(wrapper.find('i').prop('title')).toBe('Alert icon');

    expect(wrapper).toMatchSnapshot();
  });
});
