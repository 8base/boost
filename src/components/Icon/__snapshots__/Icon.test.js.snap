// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should shallow input 1`] = `
<Icon
  color="DANGER"
  name="Alert"
  size="md"
  title="Alert icon"
>
  <Boost(iconWrapper)
    color="DANGER"
    size="md"
    tagName="span"
  >
    <Tag
      className="css-rkvmzw-StyledTag emotion-0"
      color="DANGER"
      size="md"
      tagName="span"
    >
      <span
        className="css-rkvmzw-StyledTag emotion-0"
      >
        <Boost(iconSvg)
          modifiers={
            Object {
              "color": "DANGER",
              "size": "md",
            }
          }
          tagName="i"
          title="Alert icon"
        >
          <Tag
            className="css-rkvmzw-StyledTag emotion-0"
            modifiers={
              Object {
                "color": "DANGER",
                "size": "md",
              }
            }
            tagName="i"
            title="Alert icon"
          >
            <i
              className="css-rkvmzw-StyledTag emotion-0"
              title="Alert icon"
            >
              <Alert
                height="100%"
                viewBox="0 0 35 31"
                width="100%"
              >
                <svg
                  height="100%"
                  viewBox="0 0 35 31"
                  width="100%"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                  >
                    <path
                      d="M2.147 30.005c-1.913 0-2.696-1.319-1.739-2.931L15.761 1.209c.957-1.612 2.523-1.612 3.48 0l15.354 25.865c.957 1.612.174 2.931-1.74 2.931H2.147z"
                      fill="currentColor"
                    />
                    <text
                      fill="#FFF"
                      fontFamily="Poppins-Bold, Poppins"
                      fontSize="18"
                      fontWeight="bold"
                    >
                      <tspan
                        x="14"
                        y="25"
                      >
                        !
                      </tspan>
                    </text>
                  </g>
                </svg>
              </Alert>
            </i>
          </Tag>
        </Boost(iconSvg)>
      </span>
    </Tag>
  </Boost(iconWrapper)>
</Icon>
`;
